x-pack/plugins/code/public/components/admin_page/project_tab.tsx

```diff
@@ -131,15 +133,22 @@ class CodeProjectTab extends React.PureComponent<Props, State> {
   public onChange = (e: ChangeEvent<HTMLInputElement>) => {
     this.setState({
       repoURL: e.target.value,
+      isInvalid: isImportRepositoryURLInvalid(e.target.value),
     });
   };
 
   public submitImportProject = () => {
     if (!isImportRepositoryURLInvalid(this.state.repoURL)) {
       this.props.importRepo(this.state.repoURL);
+    } else if (!this.state.isInvalid) {
+      this.setState({ isInvalid: true });
```

Comment:
I think when you are trying to submit the project, the `isInvalid` should have already been correctly set by the `onChange`?. I feel like the logic here for the entire function should be

```
if (!this.state.isInvalid) {
  this.props.importRepo(this.state.repoURL);
}
```

comments?

Reply:
if you didn't input anything and click `import`, `isInvalid` is actually `false`.

Reply:
ok, make sense. then I think it worth a comment in here.